clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerRebalanceListenerInvoker.java

```diff
@@ -66,15 +66,15 @@ Exception invokePartitionsAssigned(final SortedSet<TopicPartition> assignedParti
                 throw e;
             } catch (Exception e) {
                 log.error("User provided listener {} failed on invocation of onPartitionsAssigned for partitions {}",
-                        listener.getClass().getName(), assignedPartitions, e);
+                        listener.get().getClass().getName(), assignedPartitions, e);
```

Comment:
Should we get this change in a separate PR as this touches the legacy consumer as well?

Reply:
Will do.

Reply:
Done. See the new PR #14931.